### PR TITLE
Hygiene webapp improvements

### PR DESCRIPTION
**Pull request by davemasselink**
My goal for this round of test improvement was >50% coverage across project and at least 20 tests.

Full app level "smoke tests" including routing and then good coverage in GraphContainer queryActions is what I've invested in, so far.

Haven't yet built out testing for: MenuBar, Graph, Notifier, Results

screen shot 2018-12-18 at 8 34 05 pm

### CODE REVIEW

File: app/src/components/home/Home.test.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  let input = wrapper.find('input').first();
+  input.simulate('change', { target: { value: SEARCH_STR}});
+  //and try to submit the query
+  input.simulate('keypress', {key: 'Enter'});
+  //check for expected state of the store
+  const nowStore = store.getState();
+  expect(nowStore.query.current).toEqual(SEARCH_STR);
+  expect(nowStore.query.submitted).toEqual(false);
+  expect(nowStore.query.isWaiting).toEqual(false);
+  //also check for existience of Notification
```
Comment (ssmails) [inline app/src/components/home/Home.test.js]: Is there a specific way to check the notification here ?

**Comment by ssmails** [comment]
Nice! good start

File: app/src/components/home/Home.test.js (hunk 1 above)
Comment (davemasselink) [reply app/src/components/home/Home.test.js]: yes, there is a way but it's not implemented yet

really, that check may be better as a separate test (one checking data, the other looking for DOM node creation)

I'll mark that comment as a TODO to come back to soon and will drop a few of my quick thoughts there, as well

**Comment by davemasselink** [comment]
@ssmails, thanks for taking a look. yes, I should have indicated in my description that I haven't really built out testing yet for: MenuBar, Graph, Notifier, Results

But full app level "smoke tests" including routing and then good coverage in GraphContainer queryActions is what I've invested in, so far.

**Comment by davemasselink** [comment]
@arkxu, thanks for taking a look... please merge when you're able

it's true that entity API/graph component data is being requested out of an action dispatch, that's because that component hasn't been updated to use redux store yet

the data request to the query API, however, is being triggered in the queryActions and this is the pattern we'll continue to use moving forward

**Pull request merged**